### PR TITLE
feat: add invisible Google reCAPTCHA

### PR DESCRIPTION
**Pull request by raisedadead**
Client side implementation for https://github.com/freeCodeCamp/freeCodeCamp/pull/35677

### CODE REVIEW

File: src/components/Donation/components/CardForm.js [hunk 1]
```diff
@@ -35,21 +51,32 @@ class CardForm extends PureComponent {
...
           bsStyle='primary'
-          disabled={!isFormValid}
+          disabled={!isFormValid && !isCaptchaReady}
           id='confirm-donation-btn'
           type='submit'
           >
           {`Confirm your donation of $${amount} / month`}
         </Button>
       </form>
+      <div style={{display: 'none'}}>
```
Comment (ValeraS) [inline src/components/Donation/components/CardForm.js]: Do we need this `div`?

File: src/components/Donation/components/CardForm.js [hunk 2]
```diff
@@ -35,21 +51,32 @@ class CardForm extends PureComponent {
...
     return (
+      <Fragment>
       <form className='donation-form' onSubmit={this.submit}>
         <StripCardForm getValidationState={this.getValidationState} />
         <Button
           block={true}
           bsSize='lg'
           bsStyle='primary'
-          disabled={!isFormValid}
+          disabled={!isFormValid && !isCaptchaReady}
```
Comment (ValeraS) [inline src/components/Donation/components/CardForm.js]:
```suggestion
          disabled={!isFormValid || !isCaptchaReady}
```

File: src/components/Donation/components/CardForm.js [hunk 3]
```diff
@@ -14,16 +15,31 @@ class CardForm extends PureComponent {
...
+      isFormValid: false,
+      isCaptchaReady: false
     };
 
+    this.captcha = React.createRef();
+
     this.getValidationState = this.getValidationState.bind(this);
     this.submit = this.submit.bind(this);
+    this.onVerify = this.onVerify.bind(this);
+    this.onLoad = this.onLoad.bind(this);
```
Comment (ValeraS) [inline src/components/Donation/components/CardForm.js]: do not need `bind` arrow functions.

File: src/components/Donation/Donation.js [hunk 4]
```diff
@@ -109,7 +109,18 @@ class DonationModal extends PureComponent {
                 renderCompletion={this.renderCompletion}
               />
             </Modal.Body>
-            <Modal.Footer>
+            <Modal.Footer className='GoogleDisclaimer'>
+              <h6>
+                This site is protected by reCAPTCHA and the Google
+                <a href='https://policies.google.com/privacy'>
```
Comment (ValeraS) [inline src/components/Donation/Donation.js]: Should it be with `target='_blank'`?

File: src/components/Donation/components/CardForm.js (hunk 1 above)
Comment (raisedadead) [reply src/components/Donation/components/CardForm.js]: The `reaptcha` library doesn't deem to behave if I passed this style in the `Reaptcha` component.

Comment (ValeraS) [reply src/components/Donation/components/CardForm.js]: `size=invisible` makes a `reCapture` invisible, doesn't it?

Comment (raisedadead) [reply src/components/Donation/components/CardForm.js]: Invisible is not going to get rid of the automatic badge that Google is injecting
https://developers.google.com/recaptcha/docs/versions

This is about hiding the ugly badge on the bottom corner on the page. Also, there is a complexity of the modal making it even more ugly.

The recommended way is to hide the badge that this component exposes (by first making it inline) and then adding the style.

Finally we can add the disclaimer about the privacy terms as mandated by following this:
https://developers.google.com/recaptcha/docs/faq

Yup, this is ugly. But we need it.

File: src/components/Donation/components/CardForm.js (hunk 3 above)
Comment (raisedadead) [reply src/components/Donation/components/CardForm.js]: Hmm. Thanks. I was following the pattern, and thought all this was there for some reason. TIL something.

File: src/components/Donation/Donation.js (hunk 4 above)
Comment (raisedadead) [reply src/components/Donation/Donation.js]: Ah, nice catch. Thanks.

**Pull request merged**